rm/rm-server/src/test/java/org/ow2/proactive/resourcemanager/frontend/topology/TestTopologyConcurrency.java

```diff
@@ -0,0 +1,186 @@
+package org.ow2.proactive.resourcemanager.frontend.topology;
+
+import org.apache.commons.lang3.SystemUtils;
+import org.apache.log4j.BasicConfigurator;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.Test;
+import org.objectweb.proactive.core.ProActiveException;
+import org.objectweb.proactive.core.UniqueID;
+import org.objectweb.proactive.core.body.UniversalBody;
+import org.objectweb.proactive.core.descriptor.data.ProActiveDescriptorInternal;
+import org.objectweb.proactive.core.descriptor.data.VirtualNodeInternal;
+import org.objectweb.proactive.core.descriptor.services.TechnicalService;
+import org.objectweb.proactive.core.filetransfer.FileTransferEngine;
+import org.objectweb.proactive.core.jmx.mbean.ProActiveRuntimeWrapperMBean;
+import org.objectweb.proactive.core.jmx.notification.GCMRuntimeRegistrationNotificationData;
+import org.objectweb.proactive.core.jmx.server.ServerConnector;
+import org.objectweb.proactive.core.mop.ConstructorCall;
+import org.objectweb.proactive.core.mop.ConstructorCallExecutionFailedException;
+import org.objectweb.proactive.core.node.Node;
+import org.objectweb.proactive.core.node.NodeException;
+import org.objectweb.proactive.core.node.NodeImpl;
+import org.objectweb.proactive.core.process.UniversalProcess;
+import org.objectweb.proactive.core.runtime.ProActiveRuntime;
+import org.objectweb.proactive.core.runtime.ProActiveRuntimeImpl;
+import org.objectweb.proactive.core.runtime.VMInformation;
+import org.objectweb.proactive.core.util.ProActiveInet;
+import org.ow2.proactive.resourcemanager.core.properties.PAResourceManagerProperties;
+import org.ow2.proactive.resourcemanager.frontend.topology.pinging.HostsPinger;
+import org.ow2.proactive.resourcemanager.rmnode.RMNode;
+import org.ow2.proactive.resourcemanager.selection.SelectionManager;
+import org.ow2.proactive.resourcemanager.selection.topology.TopologyManager;
+import org.ow2.proactive.scripting.ScriptResult;
+import org.ow2.proactive.scripting.SelectionScript;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.net.InetAddress;
+import java.rmi.AlreadyBoundException;
+import java.rmi.dgc.VMID;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test add and remove nodes from multiple threads
+ * <p/>
+ * At the end, remove all previously added nodes and verify that
+ * the topology manager structure is empty
+ */
+public class TestTopologyConcurrency {
+
+    // number of thread used in this test
+    final int nb_threads = 20;
```

Comment:
"Test" should be always at the end of the name of the test class and at the beginning of the test methods :)